cl/alerts/tasks.py

```diff
@@ -44,12 +43,15 @@ class DocketAlertRecipient:
 def get_docket_alert_recipients(
     d_pk: int,
     recap_email_recipients: list[str],
+    for_recap_email_user: bool = False,
```

Comment:
This is just a performance enhancement when we know there are no subscribers, right? If so, we should make it more clear in the docstring and probably in the arg name. Maybe call it `recap_email_user_only`, instead, and add something to the docstring that says something like:

> If this is a new docket, we know that there are no subscribers, and we can save queries by saying so. Set this to true to only send alerts for the recap.email user, without checking for subscribers in the DB.

Reply:
Well yes, this works as a performance improvement as a side effect but it’s not its main purpose.

I've renamed it to `recap_email_user_only`, it makes sense.

We use this bool to avoid sending duplicated docket alerts to users that are already subscribed to the docket when we receive a recap.email notification two or more times for the same docket entry (when two or more recap.email users added their recap.email address to the same case) since we expect to receive one email for each recap.email user.

So the first time we receive the recap.email notification we add the `DocketEntry`/`RECAPDocument` so `content_updated` is `True` we know this is a new `DocketEntry` and we should send the docket alert to all the users that are subscribed to the docket and also send the first email to the recap.email user from who we received the notification (user will be subscribed if the auto-subscribe option is enabled otherwise will create an unsubscription docket alert).

Then if we receive a second or more notifications for the same docket entry from other recap.email users. This time `content_updated` it’s `False` so now there are no new docket entries to trigger the docket alert. That’s why I added the new return `des_returned` to `add_docket_entries` method so even though there are no new docket entries we’ll have a list of the docket entries related to the notification so we use these docket entries list to send the docket alert only to the recap.email user from who we received the repeated notification.

That’s why `recap_email_user_only` is only True if `docket_entries` is not empty.

Reply:
OK, so put another way, we have three users:

 - Bob: Subscribed to the case, via the "Subscribe" button on the website.
 - Atty1: Uses recap.email, and has atty1@recap.email set up in her PACER account for the case.
 - Atty2: Just started using recap.email and just added atty2@recap.email to their PACER account for the case.

An email comes in for atty1@recap.email. We:

 - Send emails to atty1@recap.email and to Bob.

Another email comes in for atty2@recap.email. We:

 - Already sent out notifications for everybody else.
 - Don't want to send additional ones.
 - Just sent a notification to atty2.

That makes sense. Later, another docket entry is filed and we get two more emails. The first is to atty2@recap.email (but the order doesn't matter). We:

 - Send emails to all subscribers, which includes atty2, Bob, and atty1.

The second email comes in to atty1@recap.email. We:

 - Do nothing.

Is that all correct? If so, can you explain this with a big comment in the code?

Reply:
Yeah! the example is all correct. I only added that in order atty1@recap.email and atty2@recap.email to receive docket alerts for future docket entries (same case) they need to have the auto-subscription option enabled, otherwise they won't receive more alerts for the case.

Added the big comment in the code using your example.